userchroot.c

```diff
@@ -313,6 +314,33 @@ static int child_fn(void* v) {
   return 0;
 }
 
+static int proc_guard(void *v) {
+
+    pid_t child_pid =
+        clone(
+            child_fn,
+            child_stack+sizeof(child_stack),
+            CLONE_NEWNS | CLONE_NEWPID | SIGCHLD,
```

Comment:
the second clone doesn't need NEWNS. Also, the mount and umount should be done within the same NS, i.e. in proc_guard. `child_fn` should just run the epilogue.